microsoftgraph/microsoft-graph-explorer-v4
# Fix: show all resources

## Overview

Fixes #3011 

GE has been calling the DevX API `openapi/tree` endpoint to get the consolidated open api tree document with both versions. This leads to a huge file. 

Currently the built in Javacript `response.json()` function truncates the content as it parses the response string to JSON. This leads to some of the children not being passed from the browser to the code hence losing some nodes along the way.

This new implementation 
- makes parallel calls to get the content of each version. 
- changes the schema so that the resources in state are grouped by the version
- removes the function that was used to filter nodes based on their version as the items are already versioned

### Demo
`security/auditLog`

![image](https://github.com/microsoftgraph/microsoft-graph-explorer-v4/assets/58787602/a0f0ca5b-ef9d-4d0c-aa6f-c4acfe12803f)

`security/auditLog/queries`

![image](https://github.com/microsoftgraph/microsoft-graph-explorer-v4/assets/58787602/7f1593f5-b06f-4770-9700-7d62fb813a9d)


### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

* Switch to beta
* In the query input, type in `security/` and confirm you get `auditLog` as one of the options
* Select `auditLog` and add a `/` to trigger a new autocomplete action and confirm you get `queries` as one of the options
